### PR TITLE
Add and Modify some options.

### PR DESCRIPTION
**Pull request by kamatama41**
The requirement at Quipper is changed.
- Replace `rate_per_method_hash` to `rate_per_host_hash`
- Add option `cookie_key`
- Add option `followlocation`
- Reject invalid method

### CODE REVIEW

File: lib/fluent/plugin/out_http_shadow.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def initialize
     config_param :username, :string, :default => nil
     config_param :password, :string, :default => nil
     config_param :rate, :integer, :default => 100
-    config_param :rate_per_method_hash, :hash, :default => nil
+    config_param :rate_per_host_hash, :hash, :default => nil
```
Comment (banyan) [inline lib/fluent/plugin/out_http_shadow.rb:29]: Just wonder adding another `config_param` is not enough?

**Comment by lamanotrama** [comment]
>  The requirement at Quipper is changed.

Could you let me know the details of this?

File: lib/fluent/plugin/out_http_shadow.rb (hunk 1 above)
Comment (kamatama41) [inline lib/fluent/plugin/out_http_shadow.rb:29]: `rate_per_host_hash` is no longer necessary at our requirement. 
I love YAGNI!

Comment (banyan) [inline lib/fluent/plugin/out_http_shadow.rb:29]: I love YAGNI, too. :smile: 
However one downside I could imagine is, 
1. There might be a bug or improvement in upstream and it's fixed in `toyama0919/fluent-plugin-http_shadow`.
2. Then if `rate_per_method_hash` is also changed, it's always hard to rebase.

**Comment by kamatama41** [comment]
We want to do request rate control not by each methods but by each proxied hosts. @lamanotrama

**Comment by lamanotrama** [comment]
I got it. thank you for explain :smiley:

File: lib/fluent/plugin/out_http_shadow.rb (hunk 1 above)
Comment (kamatama41) [inline lib/fluent/plugin/out_http_shadow.rb:29]: `rate_per_method_hash` is our original option. ref: #1 
I think that if the `1. ~` situation occurrs we can rebase.

Comment (banyan) [inline lib/fluent/plugin/out_http_shadow.rb:29]: Ah, it's done by you! https://github.com/quipper/fluent-plugin-http_shadow/pull/1
I'm okay then!

**Comment by kamatama41** [comment]
@quipper/engineering ping.

**Comment by hakobera** [comment]
:+1:

**Pull request merged**